packages/wrangler/src/dev.tsx

```diff
@@ -811,15 +818,52 @@ export async function startDev(args: StartDevOptions) {
 		}
 
 		if (config.configPath && !args.experimentalDevEnv) {
-			watcher = watch(config.configPath, {
+			configFileWatcher = watch(config.configPath, {
 				persistent: true,
 			}).on("change", async (_event) => {
-				// TODO: Do we need to handle different `_event` types differently?
-				//       e.g. what if the file is deleted, or added?
-				config = readConfig(configPath, args);
-				if (config.configPath) {
+				try {
```

Comment:
The two fns that can potentially throw here are `readConfig` and `processExperimentalAssetsArg`. Since we're in watch mode, we ideally don't want to error and exit every time there's a config file read error or we couldn't properly resolve the assets directory path. The better user experience IMO is to log the error and allow users to fix as they go, without having them re-start the dev process every single time. This brings the watch mode experience in line with `pages dev`, which we recently refactored for precisely the same reason.